esi_leap/objects/flocx_market_client.py

```diff
@@ -63,3 +64,64 @@ def send_offer(self, data):
                 r.status_code,
                 r.text)
         return r.status_code
+
+    def get_contracts(self):
+        url = CONF.flocx_market.endpoint_override + \
+        '/offer_contract_relationship?status=unretrieved'
+        r = self.get(url)
+        if r.status_code != 200:
+            LOG.warning(
+                "Failed to retrieve contracts from flocx-market. Got HTTP %s: %s",
+                r.status_code,
+                r.text)
+        if r.content is not None:
+            contracts = []
+            contract_offer_table = {}
+            contract_offer_list = json.loads(r.content)
+            # create contract offers hash table
+            # key: contract_id, value: (offer_id, offer_contract_id) list
+            for c_o in contract_offer_list:
+                if not contract_offer_table.get(str(c_o["contract_id"])):
+                    o_c_r_tuple = (str(c_o["marketplace_offer_id"]), str(c_o["offer_contract_relationship_id"]))
+                    contract_offer_table[str(c_o["contract_id"])] = [o_c_r_tuple]
+                    contract_offer_table[str(c_o["contract_id"])].append(o_c_r_tuple)
+
+            # retrieve contracts and offers data from flocx-market iteratively
+            for contract_id, o_c_rs in contract_offer_table.items():
+                contract_data = {}
+                url1 = CONF.flocx_market.endpoint_override + '/contract/' + contract_id
+                r_contract = self.get(url1)
+                c = json.loads(r_contract.content)
```

Comment:
These few lines should probably go into their own `get_contract` function

Reply:
Yeah, I moved get_contract functions to common/flocx_market.py. flocx_market_client.py does not have any contract related functions. I also created get_contract_by_id, get_ocr_by_status, get_offer_by_id in common/flocx_market.py. The code for retrieving contracts seems easier to understand :D